### ISSUE TITLE
bug(mat-slider): displayWith uses MatSlider reference as this instead of parent component

### ISSUE THREAD

**Issue by sherif-elmetainy** [angular/components#22110]
#### Reproduction

Use StackBlitz to reproduce your issue: 
- Components: https://stackblitz.com/edit/components-issue-wb3oeb

Steps to reproduce:
1. User mat-slider in a component
2. Define a function in the hosting component that computes the displayValue
    Example: 
    ```typescript
    export class ExampleComponent {
         public percentSign = "%";
         formatValue(n: number): string {
             // this keyword in the next line refences the MatSlider component rather than ExampleComponent
            return n.toString() + ' '+ this.percentSign;
       }
    }
    ```
3. Set the `displayWith` value to the defined function
    Example:
    ```html
    <mat-slider min="0" max="100" [displayWith]="formatValue" thumbLabel="true"></mat-slider>
    ```
 
 
#### Expected Behavior
I expected when the formatValue function is called, that `this` keyword would refer to the ExampleComponent reference. And hence, the `this.percentSign` would be '%'.

#### Actual Behavior
When debugging the formatValue function is called, `this` keyword refers the MatSlider component reference instead of the host component. Since MatSilder doesn't have pecentSign property, it is displayed as undefined.

Note: My actual implementation access other properties in the component so having access to my component via `this` keyword is important.

#### Environment

- Angular: 11.2.3
- CDK/Material: 11.2.2
- Browser(s): Google Chrome Version 88.0.4324.190
- Operating System (e.g. Windows, macOS, Ubuntu): Windows 10

**Comment by jelbourn** [comment]
This is working as expected in that `MatSlider` doesn't do anything with the identity of `this`- `MatSlider` would have no way of having a reference to your component that contains the slider. 

In a general JavaScript sense, the expectation is that the function you're passing is either an arrow function or is bound via `.bind`. Whenever you take a JavaScript function that depends on `this`, passing that function into _any_ API without binding it (or using an arrow function) will drop the this context and the behavior becomes undefined.

**Comment by sherif-elmetainy** [comment]
@jelbourn I understand your point and I actually solved it using `bind` right after I created the issue. 

However, I disagree with you that this is the expectation. Having use this dependent functions all the time in angular, for example `(click)=` on a button, or `(dragEnter)=` on any element, etc. But the property is also inconsistent with all callbacks like click, etc in that it's not an `@Output` property.

Anyway, thanks for taking the time to look at the issue.
